src/app/meshes/components/MeshDetails.vue

```diff
@@ -2,95 +2,156 @@
   <div class="stack">
     <KCard>
       <template #body>
-        <MeshCharts :mesh-insight="meshInsight" />
-      </template>
-    </KCard>
+        <div class="stack">
+          <div
+            class="columns"
+            style="--columns: 4"
+          >
+            <DefinitionCard>
+              <template #title>
+                {{ t('http.api.property.name') }}
+              </template>
 
-    <KCard>
-      <template #body>
-        <div class="columns">
-          <DefinitionList>
-            <DefinitionListItem
-              v-for="(value, property) in basicMesh"
-              :key="property"
-              :term="t(`http.api.property.${property}`)"
+              <template #body>
+                <TextWithCopyButton :text="props.mesh.name">
+                  <RouterLink
+                    :to="{
+                      name: 'mesh-detail-view',
+                      params: {
+                        mesh: props.mesh.name,
+                      },
+                    }"
+                  >
+                    {{ props.mesh.name }}
+                  </RouterLink>
+                </TextWithCopyButton>
+              </template>
+            </DefinitionCard>
+
+            <ResourceStatus
+              :total="props.meshInsight.services.total ?? 0"
+              data-testid="services-status"
             >
-              <KBadge
-                v-if="typeof value === 'boolean'"
-                :appearance="value ? 'success' : 'danger'"
-              >
-                {{ value ? 'Enabled' : 'Disabled' }}
-              </KBadge>
-
-              <template v-else-if="property === 'name' && typeof value === 'string'">
-                <TextWithCopyButton :text="value" />
+              <template #title>
+                {{ t('meshes.detail.services') }}
               </template>
+            </ResourceStatus>
 
-              <template v-else>
-                {{ value }}
+            <ResourceStatus
+              :total="props.meshInsight.dataplanesByType.standard.total ?? 0"
```

Comment:
can any of these be null when a dp type does not exist (e.g. gateway)? should we do optional chaining?

Reply:
Yes. Specifically, `total` can be absent here. So while `dataplanesByType` always exists and always has the properties `standard` and `gateway`, these objects may be completely empty.

Reply:
I misread your question. You’re asking if `gateway` and/or `standard` can be absent, and I don’t think they can.